### PR TITLE
Tabbed Content Widget (Issue #518)

### PR DESCRIPTION
**Pull request by dtsnell4**
Basic layout of tabs and content panels

Signed-off-by: Dan Snell Dan.Snell@lexisnexis.com

Fixes TabbedContent serialization and streamlines groupClass concept (For uniquely identifying tabs/content belonging to a specific instance of TabbedContent when nesting instances of TabbedContent)

Signed-off-by: Jay Brundage jaman.brundage@lexisnexis.com

### CODE REVIEW

**Comment by dtsnell4** [comment]
@GordonSmith @jbrundage @mlzummo Please Review

File: demos/dermatology.html [hunk 1]
```diff
@@ -68,13 +68,11 @@
...
-                            <option value="src/chart/Pie">Pie</option>
-                            <option value="src/chart/Bubble">Bubble</option>
                             <option value="src/chart/Column">Column</option>
-                            <option value="src/chart/Scatter">Scatter</option>
+                            <option value="src/chart/Bubble">Bubble</option>
                             <option value="src/chart/Line">Line</option>
-                            <option value="src/chart/Area">Area</option>
-                            <option value="src/chart/Step">Step</option>
+                            <option value="src/chart/Scatter">Scatter</option>
+                            <option value="src/chart/Pie">Pie</option>
```
Comment (GordonSmith) [inline demos/dermatology.html]: I am not sure these should be changing?

File: demos/dermatology.html [hunk 2]
```diff
@@ -127,6 +126,9 @@
     <div id="main" style="width:100%; height:100vh;margin-top:40px;padding:8px;overflow:hidden">
     </div>
     <script>
+        var g_widget;
+        var g_persist;
+        var g_serial;
```
Comment (GordonSmith) [inline demos/dermatology.html]: These are assigned to, but not used?

File: demos/dermatology.html [hunk 3]
```diff
@@ -94,6 +92,7 @@
                             <option value="src/layout/Cell">Cell Layout</option>
                             <option value="src/layout/Grid">Grid Layout</option>
                             <option value="src/layout/Border">Border Layout</option>
+                            <option value="src/layout/TabbedContent">Tabbed Layout</option>
```
Comment (GordonSmith) [inline demos/dermatology.html]: I am not sure the other changes in this file are wanted?

File: src/other/Persist.js [hunk 4]
```diff
@@ -109,8 +109,8 @@
                                 });
                                 break;
                             case "widgetArray":
-                                ++createCount;
                                 var widgetArrayKey = key;
+                                ++createCount;
```
Comment (GordonSmith) [inline src/other/Persist.js]: This change looks unintentional?

File: demos/dermatology.html (hunk 2 above)
Comment (dtsnell4) [inline demos/dermatology.html]: @jbrundage just checking, these are no longer necessary correct?

File: src/other/Persist.js (hunk 4 above)
Comment (dtsnell4) [inline src/other/Persist.js]: @jbrundage do you move this for a reason?  I cannot recall.

**Comment by GordonSmith** [comment]
Closing in preference to:  https://github.com/hpcc-systems/Visualization/pull/633

**Pull request closed without merging**